webapp/src/monaco.tsx

```diff
@@ -787,31 +787,34 @@ export class Editor extends srceditor.Editor {
 
             let snippetPrefix = fn.noNamespace ? "" : ns;
             let isInstance = false;
+            let addNamespace = false;
 
             const element = fn as pxtc.SymbolInfo;
             if (element.attributes.block) {
                 if (element.attributes.defaultInstance) {
                     snippetPrefix = element.attributes.defaultInstance;
                 }
-                else if (element.kind == pxtc.SymbolKind.Method || element.kind == pxtc.SymbolKind.Property) {
-                    const params = pxt.blocks.parameterNames(element);
-                    snippetPrefix = params.attrNames["this"].name;
-                    isInstance = true;
-                }
                 else if (element.namespace) { // some blocks don't have a namespace such as parseInt
                     const nsInfo = this.blockInfo.apis.byQName[element.namespace];
-                    if (nsInfo.kind === pxtc.SymbolKind.Class) {
-                        return undefined;
-                    }
-                    else if (nsInfo.attributes.fixedInstances) {
+                    if (nsInfo.attributes.fixedInstances) {
                         const instances = Util.values(this.blockInfo.apis.byQName).filter(value =>
                             value.kind === pxtc.SymbolKind.Variable &&
                             value.attributes.fixedInstance &&
-                            value.retType === nsInfo.name)
+                            value.retType.endsWith(nsInfo.name))
```

Comment:
Don't use `endsWith` (not all browsers support it). I think we have `Util.endsWith()` that you can use instead